### ISSUE TITLE
Could PEcAn be used with non-site-oriented model?

### ISSUE THREAD

**Issue by bpbond** [PecanProject/pecan#1813]
Hi all,

I asked @mdietze this yesterday, but didn't convey my question clearly.

Would it be possible, with some moderate amount of work, to take advantage of all of the PEcAn code for parameter sensitivity, MCMC, etc., for a non-site oriented model? I'm thinking in particular of [Hector](https://github.com/JGCRI/hector), a global-scale carbon and climate model, with simple (text file) inputs and outputs, that produces estimates of the regional to global C cycle, ocean chemistry and fluxes, radiative forcing, etc. We have some basic R code that can do sensitivity analyses for it, but it would be crazily fantastic to take advantage of the PEcAn interface and sophistication.

I recognize that it may be too far removed from the kinds of models PEcAn was designed for, but thought I'd ask.

Thanks!

**Comment by mdietze** [comment]
Yes. If that's the default mode for the model then you could always set it up in PEcAn and then run it with the whole earth as a the site (*). PEcAn wouldn't be prepping the inputs, and we may need to do some tweaks on how it handles the outputs (since they'd all be space-time not timeseries), but I don't think it'd be hard. 

* because PEcAn uses PostGIS under the hood a "site" in the pecan database could be any geometry that PostGIS supports -- point, bounding box, polygon, line, etc.  There's a few 'bounding box' sites in the database already, but so far nothing more complex than that.

**Comment by bpbond** [comment]
Thanks. Wow, that sounds promising! I'm going to open an issue over in Hector for us to think about this.

**Comment by mdietze** [comment]
For reference the Earth is site.id 1160 and just the land surface is site.id 1118. We also have boxes for things like the spatial extent of a few regional datasets (e.g NARR)

**Comment by ashiklom** [comment]
Re-branding this issue as being explicitly about Hector.

**Comment by dlebauer** [comment]
Following on Mike's comment, the sensitivity analysis should be able to work on any model output. If you have an output like 'global soil carbon stocks'  with just a time dimension it should fit the current workflow. If you want to run the analysis for every point on a grid or otherwise account for spatial structure, it could take more work (and thought).

I've used PEcAn to do regional runs in BioCro with some modifications to the BioCro executable and pre-generated met and soil files. It was fairly straightforward.
